test/honey/util_test.cljc

```diff
@@ -43,3 +43,20 @@
   (is (= "1, 2, 3, 4"
          (sut/join ", " (remove nil?) [1 nil 2 nil 3 nil nil nil 4])))
   (is (= "" (sut/join ", " (remove nil?) [nil nil nil nil]))))
+
+(deftest split-by-separator-test
```

Comment:
I'd feel a bit more comfortable if these tests also compared against `(str/split <string> #"\.")` just to verify they're equivalent to that (slower) form that is being replaced in the source.